Filtering submissions for system errors raises a system error

Steps to reproduce:
1. Create/go to any contest
2. View the submissions list in the "Contest administration" navbar category
3. Filter with "Has active system error" set to yes/no
4. Experience a 500

Example url: ``/c/a/admin/contests/submission/?has_active_system_error=no``.

<details>
<summary>Traceback (most recent call last):</summary>
<div class="highlight">
<pre tabindex="0" class="chroma">
<code class="language-fallback" data-lang="fallback">
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/options.py", line 616, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/sites.py", line 232, in inner
    return view(request, *args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/options.py", line 1697, in changelist_view
    cl = self.get_changelist_instance(request)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/options.py", line 749, in get_changelist_instance
    sortable_by,
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/views/main.py", line 99, in __init__
    self.queryset = self.get_queryset(request)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/contrib/admin/views/main.py", line 477, in get_queryset
    request, qs, self.query,
  File "/sio2/oioioi/oioioi/contests/admin.py", line 706, in get_search_results
    problem_instance__problem__in=matched_problems
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/db/models/query.py", line 345, in __or__
    combined.query.combine(other.query, sql.OR)
  File "/home/oioioi/.local/lib/python3.7/site-packages/django/db/models/sql/query.py", line 591, in combine
    "Cannot combine a unique query with a non-unique query."
AssertionError: Cannot combine a unique query with a non-unique query.
</code></pre></div></details>

--- a/oioioi/contests/admin.py
+++ b/oioioi/contests/admin.py
@@ -633,9 +633,9 @@ class SystemErrorListFilter(SimpleListFilter):
             submissionreport__status='ACTIVE', submissionreport__testreport__status='SE'
         )
         if self.value() == 'yes':
-            return queryset.filter(q).distinct()
+            return queryset.filter(q)
         elif self.value() == 'no':
-            return queryset.exclude(q).distinct()
+            return queryset.exclude(q)
         else:
             return queryset
 

--- a/oioioi/contests/tests/tests.py
+++ b/oioioi/contests/tests/tests.py
@@ -245,6 +245,37 @@ class TestSubmissionListOrder(TestCase):
         self.assertTrue(test_first_index < test_second_index, error_msg)
 
 
+# TODO: expand this TestCase
+class TestSubmissionListFilters(TestCase):
+    fixtures = [
+        'test_users',
+        'test_contest',
+        'test_full_package',
+        'test_problem_instance',
+        'test_submission',
+        'test_another_submission',
+        'test_submissions_CE',
+    ]
+
+    def setUp(self):
+        self.assertTrue(self.client.login(username='test_admin'))
+        self.url = reverse(
+            'oioioiadmin:contests_submission_changelist', kwargs={'contest_id': 'c'}
+        )
+
+    def test_all_filters(self):
+        response = self.client.get(self.url, {
+            'has_active_system_error': 'yes',
+            'kind': 'NORMAL',
+            'status__exact': 'INI_OK',
+            'revealed': '1',
+            'round': 'Round 1',
+            'lang': 'Pascal',
+        })
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, '0 submissions')
+
+
 @override_settings(
     CONTEST_MODE=ContestMode.neutral, ROOT_URLCONF='oioioi.contests.tests.test_urls'
 )

